src/ripple/protocol/impl/Feature.cpp

```diff
@@ -20,17 +20,43 @@
 #include <BeastConfig.h>
 #include <ripple/protocol/digest.h>
 #include <ripple/protocol/Feature.h>
+
+#include <boost/container/flat_map.hpp>
+
 #include <cstring>
 
 namespace ripple {
 
+static boost::container::flat_map<uint256, std::size_t> featureToIndex;
+static std::vector<uint256> features;
```

Comment:
@mellery451 I agree with your analysis of the intended use of features.

All the features in rippled should live in Features.h and follow the pattern
we're using there (global variable named after the feature). Is there some weird
use-case I'm missing? If so, let's look closely at it to see if we can resolve
that use-case. If not, I have some ideas to run past y'all to address both
thread safety and the asserts brought up by @scottschurr 

1) Thread safety. 

How about keeping a constexpr array of feature names. When adding a new feature,
a new name is added to this array and the size increased. We can add an
initializers that run before main and create the collections we need to support
the FeatureBitset. We'll also move the family of `uint256 feature(std::string)`
functions to the `cpp` file, so they can only be used when first creating
features.

Since the FeatureBitset supporting collections will be `const`, they will be
type safe when querying them and we won't need to use locks.

We'll also add a `boost::optional<uint256> getRegisteredFeature (std::string
const& name);` function for the places that were using the old
`feature(std::string)` interface (config.cpp and a couple tests). Note, config
will now throw if the config contains an unregistered feature. I think that's a
good thing, but others should weigh in, of course.

2) Asserts.

Since we know the number of features at compile time, the asserts go away, and
the `FeatureBitset` size will always equal the number of features.

BTW, I considered using a variatic template to initialize the collection of
feature names. That would make it _slightly_ easier to add a new feature, as the
number of features could be computed rather than needing to increment it.
However, I think the simplicity of just declaring the array directly is better,
but something to consider.

Here's a commit to consider. Notice this _does_ change behavior (config in particular).
https://github.com/seelabs/rippled/commit/a3b0cba139e5b397cd8bef87257ceca8d340cca9